connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/entities/ConnectorOffset.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.connect.runtime.rest.entities;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Represents a single {partition, offset} pair for either a sink connector or a source connector. For source connectors,
+ * the partition and offset structures are defined by the connector implementations themselves. For a sink connector,
+ * where offsets represent the underlying Kafka consumer group offsets, this would look something like:
+ * <pre>
+ *     {
+ *       "partition": {
+ *         "kafka_topic": "topic"
+ *         "kafka_partition": 3
+ *       },
+ *       "offset": {
+ *         "kafka_offset": 1000
+ *       }
+ *     }
+ * </pre>
+ */
+public class ConnectorOffset {
+    public static final String KAFKA_TOPIC_KEY = "kafka_topic";
+    public static final String KAFKA_PARTITION_KEY = "kafka_partition";
+    public static final String KAFKA_OFFSET_KEY = "kafka_offset";
```

Comment:
Nit: I find it a little strange that this class is almost entirely agnostic about source vs. sink offsets, but it defines these sink-specific constants. Maybe they would make more sense in the `SinkUtils` class, or we could add a static `forSink(Map<TopicPartition, Long>)` (or `OffsetAndMetadata` instead of `Long` if that's easier) factory method to this class that replaces `SinkUtils::consumerGroupOffsetsToConnectorOffsets`?